lib/media/track/localmediatrack.js

```diff
@@ -103,16 +106,72 @@ function mixinLocalMediaTrack(AudioOrVideoTrack) {
       }
       super._initialize.call(this);
     }
+
     /**
      * @private
      */
-    _end() {
-      if (this._didCallEnd) {
-        return;
-      }
-      super._end.call(this);
-      this._didCallEnd = true;
-      this.emit('stopped', this);
+    _reacquireTrack(constraints) {
+      const {
+        _getUserMedia: getUserMedia,
+        _log: log,
+        mediaStreamTrack: { kind }
+      } = this;
+
+      log.info('Re-acquiring the MediaStreamTrack');
+      log.debug('Constraints:', constraints);
+
+      const gUMConstraints = Object.assign({
+        audio: false,
+        video: false
+      }, { [kind]: constraints });
+
+      return getUserMedia(gUMConstraints).then(mediaStream => {
+        return mediaStream.getTracks()[0];
+      });
+    }
+
+    /**
+     * @private
+     */
+    _restart(constraints) {
+      const { _log: log, mediaStreamTrack } = this;
+      constraints = constraints || mediaStreamTrack.getConstraints();
+      return this._reacquireTrack(constraints).catch(error => {
+        log.error('Failed to re-acquire the MediaStreamTrack:', error, constraints);
+        throw error;
+      }).then(newMediaStreamTrack => {
+        log.info('Re-acquired the MediaStreamTrack');
+        log.debug('MediaStreamTrack:', newMediaStreamTrack);
+        return this._setMediaStreamTrack(newMediaStreamTrack);
```

Comment:
we should to catch and log errors from `_setMediaStreamTrack` and not propagate them.

Reply:
We already do that [here](https://github.com/twilio/twilio-video.js/pull/1062/files#diff-65a9cef12e9eb87a881dd86ac35ad263R160).